### DISCUSSION TITLE
Ecommerce framework Bundle

### DISCUSSION THREAD

**Discussion by creativehubspace** [pimcore/pimcore#15988] [Q&A / Support]
I have configured the E-commerce Framework with mostly the default config settings and I can successfully add products to cart. However I noticed an issue. The session cart is being mixed up across multiple visitors.  For instance, if I add products to cart as a guest user in chrome, and then open the website in Firefox or in incognito mode in chrome, the cart items are also getting loaded in these browser instances. I expect the cart to be empty here, however, that is not the case. 

I am using Pimcore 10.5.25

Is there something that I am missing?

**Comment by dvesh3** [comment]
Hi @creativehubspace please make sure that you don't have a session based on same logged in user on both browser otherwise I don't see this as a valid scenario. maybe you can check if [fullpage cache ](https://pimcore.com/docs/platform/Pimcore/Development_Tools_and_Details/Cache/Full_Page_Cache#configure-the-full-page-cache)is not interfering somehow?

**Comment by creativehubspace** [comment]
Hi @dvesh3 Thank you for your response. I am logged out in both browser sessions. Fullpage cache is also disabled. The value of the **PHPSESSID** on both browsers is also different.

**Reply by dvesh3** [reply]
could you please try on pubic demo https://demo.pimcore.com once? just to get confirmation that it is not specific to your system. As i cannot reproduce there.

**Reply by creativehubspace** [reply]
I can't reproduce it in the demo either. However I have noticed something. If I clear the value of **PHPSESSID** for the demo in the browser, and reload the page, the cart is emptied. This is not the case in my system. When I do the same in my system, the cart is still loaded with the cart items.

**Comment by creativehubspace** [comment]
I had to set the environment to use the guest cart when not logged in. Added this code in one of my controllers.
`
$environment = Factory::getInstance()->getEnvironment();
    $user = $this->getUser();
    if (!$user) {
        $environment->setUseGuestCart(true);
        $environment->save();
    }
`
